go/storage/database/database.go

```diff
@@ -164,3 +177,59 @@ func (ba *databaseBackend) Checkpointer() checkpoint.CreateRestorer {
 func (ba *databaseBackend) NodeDB() dbApi.NodeDB {
 	return ba.ndb
 }
+
+// autoDetectBackend attempts automatic backend detection, modifying the configuration in place.
+func autoDetectBackend(cfg *api.Config) error {
+	if cfg.Backend != BackendNameAuto {
+		return nil
+	}
+
+	// Make sure that the DefaultFileName was used to derive the subdirectory. Otherwise automatic
+	// detection cannot be performed.
+	if filepath.Base(cfg.DB) != DefaultFileName(cfg.Backend) {
+		return fmt.Errorf("storage/database: 'auto' backend selected using a non-default path")
+	}
+
+	// Perform automatic database backend detection if selected. Detection will be based on existing
+	// database directories. If multiple directories are available, the most recently modified is
+	// selected.
+	type foundBackend struct {
+		path      string
+		timestamp time.Time
+		name      string
+	}
+	var backends []foundBackend
+
+	for _, b := range db.Backends {
+		// Generate expected filename for the given backend.
+		fn := DefaultFileName(b.Name())
+		maybeDb := filepath.Join(filepath.Dir(cfg.DB), fn)
+		fi, err := os.Stat(maybeDb)
+		if err != nil {
+			continue
+		}
+
+		backends = append(backends, foundBackend{
+			path:      maybeDb,
+			timestamp: fi.ModTime(),
+			name:      b.Name(),
+		})
+	}
+	slices.SortFunc(backends, func(a, b foundBackend) int {
+		return a.timestamp.Compare(b.timestamp)
+	})
+
+	// If no existing backends are available, default to "pathbadger".
```

Comment:
```suggestion
	// If no existing backends are available, use default.
```
We can change default in the future, which will invalidate this comment.